### DISCUSSION TITLE
Suggestion: `use_blob` default as True when user is admin

### DISCUSSION THREAD

**Discussion by MariusWirtz** [cubewise-code/tm1py#894] [Ideas]
Given the performance impact of using blobs with TM1py, it seems reasonable to use them by default, for admin users, in the following functions:

- `write`
- `execute_mdx_csv`
- `execute_view_csv`
- `execute_mdx_dataframe`
- `execute_view_dataframe`

In theory, this could be considered a breaking change in the upcoming 1.11 release. 
However, given the revolutionary performance impacts of up to 10x in write and 2x in read, it seems a reasonable decision IMO.

Any other opinions on this one?

**Comment by rclapp** [comment]
I have some reservations about this change. Moving towards blob based interaction makes  physical storage play a role in how the application scales. that seems a bit dangerous especially for applications with high concurrency.

Sent from my mobile device

On Apr 8, 2023 6:35 PM, Marius Wirtz ***@***.***> wrote:

Given the performance impact of using blobs with TM1py, it seems reasonable to use them by default, for admin users, in the following functions:

  *   write
  *   execute_mdx_csv
  *   execute_view_csv
  *   execute_mdx_dataframe
  *   execute_view_dataframe

In theory, this could be considered a breaking change in the upcoming 1.11 release.
However, given the revolutionary performance impacts of up to 10x in write and 2x in read, it seems a reasonable decision IMO.

Any other opinions on this one?

**Reply by MariusWirtz** [reply]
I understand your concern. Of course, TM1py would remove those files unless execution is interrupted, but nonetheless, it creates a new dependency on physical storage.

Possibly we could set the default for `use_blob` to `False` in 1.11 and change it to `True` (for admins) in the next major release (2.0).

**Comment by rclapp** [comment]
I think it should always be false unless specified. I think builders that use TM1py have the assumption that data read/write is more direct and relies on TM1s API endpoint built for that reason. While this pattern is faster there could be considerable tradeoffs at scale. Plus I don’t think IBM is designing their future roadmap with this “hack” (hack in the positive sense) in mind. My suggestion is that we keep it as false, then raise this performance gap with IBM and ask them to close it.

**Reply by MariusWirtz** [reply]
I find myself in agreement with your position. Let's follow that approach.
